### ISSUE TITLE
[RPC] listaddresses doesn't include mining addresses

### ISSUE THREAD

**Issue by CaveSpectre11** [Veil-Project/veil#863]
Need to look into the changes of `listaddresses` and likely add capability to scan through the mining addresses that are in the wallet.

**Comment by Rock-N-Troll** [comment]
It looks like there's a line in the `listaddresses` command:

```
// Only get basecoin and stealth addresses
        if (!((item.first.type() == typeid(WitnessV0KeyHash)) ||
              (item.first.type() == typeid(CStealthAddress)))) continue;
```

which is designed to handle CStealthAddress a few lines below:

`if (item.first.type() == typeid(CStealthAddress)) {`

so it looks like it is removed intentionally. I suspect just removing it from the continue block would fix it, but think there's some reason why it was removed in the first place

**Comment by CaveSpectre11** [comment]
> It looks like there's a line in the `listaddresses` command:
> 
> ```
> // Only get basecoin and stealth addresses
>         if (!((item.first.type() == typeid(WitnessV0KeyHash)) ||
>               (item.first.type() == typeid(CStealthAddress)))) continue;
> ```
> 
> which is strange because the code is already built to handle CStealthAddress a few lines below:
> 
> `if (item.first.type() == typeid(CStealthAddress)) {`
> 
> so it looks like it was removed intentionally. I suspect just removing it from the continue block would fix it, but think there's some reason why it was removed in the first place.
> 
> Any input would be appreciated as it seems like the fix is simple

Note that this cli command was added in PR #842, and the intent was to give a way to list the stealth addresses; so mining addresses weren't exactly considered; hence the Issue (I think).

The first check was to limit it to stealth and basecoin; the second was to get the information for stealth if stealth, else basecoin.  So your PR appears correct to add a 3rd accepted addresstype.

**Comment by seanPhill** [comment]
This issue is addressed in the merged commit of #911 `[RPC] Return mining addresses in list addresses roc command`.

(Cleaning up)
